sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractRulesDefinition.java

```diff
@@ -88,4 +106,60 @@ public void define(Context context) {
     requireNonNull(allRuleKeys);
     return allRuleKeys;
   }
+
+  private void updateMetadata(NewRule rule) {
+    if (supportsSecurityHotspots) {
+      RuleMetadata ruleMetadata = readRuleMetadata(rule.key());
+      for (String s : ruleMetadata.securityStandards.OWASP) {
+        rule.addOwaspTop10(RulesDefinition.OwaspTop10.valueOf(s));
+      }
+      rule.addCwe(ruleMetadata.securityStandards.CWE);
+      if (ruleMetadata.isSecurityHotspot()) {
+        setSecurityHotspotType(rule);
+      }
+    }
+  }
+
+  private static RuleMetadata readRuleMetadata(String ruleKey) {
+    String resourcePath = "/org/sonar/plugins/csharp/" + ruleKey + "_c#.json";
+    try (InputStream stream = AbstractRulesDefinition.class.getResourceAsStream(resourcePath)) {
+      return  stream != null
+        ? GSON.fromJson(new InputStreamReader(stream, StandardCharsets.UTF_8), RuleMetadata.class)
+        : new RuleMetadata();
+      } catch (IOException e) {
```

Comment:
[cosmetic] The alignment of the `catch` seems off.

Reply:
good catch!